es6/README.md

```diff
@@ -704,6 +704,69 @@ Button.propTypes = {
 
 Both declarative property syntaxes are not a part of the ES2015 specification and are a in the midst of the ECMAScript proposal approval process. Currently they are sitting in Stage 3. For details, check out [ECMAScript Class Fields and Static Properties](https://github.com/jeffmo/es-class-fields-and-static-properties).
 
+For bound methods within classes, we recommend to use fat arrow syntax to write functions need current context:
```

Comment:
So then are you suggesting that we not do:

```js
class Button extends React.Component {
    _handleOnClick() {
        // handle button click
    }

    render() {
        return (
             <button onClick={this._handleOnClick.bind(this)} />
        );
    }
}
```

either?

If so we'll have to update all of the other examples in this doc to reflect this new pattern.

Reply:
I think this is fine, my only intention is to replace the `::func` case because it might create some bugs where we want to refer to the same callback but `::func` basically means `func.bind` and it returns a new instance each time.

Reply:
If that's the case then, I wouldn't bother with addition because the two "bad" examples aren't code that you'll see outside of the old code you saw 😄 . Also the "good" example is different than the pattern we have been using everywhere so it'll probably cause confusion.

I think we should probably be more intention with making the change to use property arrow functions like you have here. Otherwise there'll be lots of confusion I'm afraid

Reply:
@benmvp That makes sense. Is there a place here to suggest "better" rather than using `good` and `bad`? e.g. If need to use event handlers, better to use fat arrow syntax with the reasons listed above. I feel that's the real intention from this PR.

Reply:
Or I can just replace my addition with some suggestions and put examples there.

Reply:
We can rephrase this to `When defining methods to be attached to event listeners, we recommend using the fat arrow syntax within classes to bind the current context without returning a new function each time.`

I actually prefer us moving towards this approach over the current `.bind` but don't expect anyone to refactor their old code. 

Also nit, we should use `PureComponent` instead of `Component` for consistency.

Reply:
@kaylieEB 👍 I will update my PR.

Reply:
Updated in https://github.com/eventbrite/javascript/pull/55/commits/45bd051c71a099f70b0808b3667e7b05c33388ca